src/deploy/functions/params.ts

```diff
@@ -233,32 +357,37 @@ export async function resolveParams(
 ): Promise<Record<string, ParamValue>> {
   const paramValues: Record<string, ParamValue> = {};
 
-  const [provided, outstanding] = partition(params, (param) => {
+  // TODO(vsfan@): should we ever reject param values from .env files based on the appearance of the string?
+  const [resolved, outstanding] = partition(params, (param) => {
     return {}.hasOwnProperty.call(userEnvs, param.name);
   });
-  for (const param of provided) {
-    if (!canSatisfyParam(param, userEnvs[param.name])) {
-      throw new FirebaseError(
-        "Parameter " +
-          param.name +
-          " resolved to value from dotenv files " +
-          userEnvs[param.name] +
-          " of wrong type"
-      );
-    }
-    paramValues[param.name] = userEnvs[param.name];
+  for (const param of resolved) {
+    const valueFromEnvs = Object.create(userEnvs[param.name]) as ParamValue;
```

Comment:
uh I think I need some help understanding this. ParaValue is a class, and we are casting a plain object to be a `ParamValue` object? and calling `.narrowType` on it works? Why can't we use `param` directly here?

Reply:
Wow, let's. Uh. Pretend I never wrote this code. Yeah.